sci-libs/med/med-4.0.0-r2.ebuild

```diff
@@ -0,0 +1,105 @@
+# https://github.com/Kitware/ninja/tree/features-for-fortran#readme
+CMAKE_MAKEFILE_GENERATOR=emake
+
+FORTRAN_NEEDED=fortran
+
+# NOTE:The build for multiple python versions should be possible but
+# complicated for the build system
+PYTHON_COMPAT=( python3_{5,6,7} )
+
+inherit cmake-utils fortran-2 python-single-r1
+
+DESCRIPTION="A library to store and exchange meshed data or computation results"
+HOMEPAGE="https://www.salome-platform.org/user-section/about/med"
+SRC_URI="https://files.salome-platform.org/Salome/other/${P}.tar.gz"
+
+LICENSE="LGPL-3"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="doc fortran mpi python test"
+
+REQUIRED_USE="python? ( ${PYTHON_REQUIRED_USE} )"
+# dev-lang/tk is needed for wish-based xmdump utility
+RDEPEND="
+	!sci-libs/libmed
+	dev-lang/tk:0=
+	>=sci-libs/hdf5-1.10.2:=[fortran=,mpi=]
+	mpi? ( virtual/mpi[fortran=] )
+	python? ( ${PYTHON_DEPS} )
+"
+DEPEND="${RDEPEND}"
+BDEPEND="python? ( >=dev-lang/swig-3.0.8 )"
+
+#	"${FILESDIR}/${PN}-3.3.1-mpi.patch"
+PATCHES=(
+	"${FILESDIR}/${PN}-3.3.1-cmake-fortran.patch"
+	"${FILESDIR}/${PN}-3.3.1-disable-python-compile.patch"  # managed by function of python eclass
+	"${FILESDIR}/${P}-cmakelist.patch"
+	"${FILESDIR}/${P}-tests.patch"  # disable a few tests not running
+	"${FILESDIR}/${P}-installdoc.patch"
+)
+
+DOCS=( AUTHORS ChangeLog NEWS README TODO )
+
+pkg_setup() {
+	use python && python-single-r1_pkg_setup
+	use fortran && fortran-2_pkg_setup
+}
+
+src_prepare() {
+	# fixes for correct libdir name
+	sed -i -e "s@SET(_install_dir lib/python@SET(_install_dir $(get_libdir)/python@" \
+		./python/CMakeLists.txt || die "sed failed"
+	for cm in ./src/CMakeLists.txt ./tools/medimport/CMakeLists.txt
+	do
+		sed -i -e "s@INSTALL(TARGETS \(.*\) DESTINATION lib)@INSTALL(TARGETS \1 DESTINATION $(get_libdir))@" \
+			"${cm}" || die "sed on ${cm} failed"
+	done
+
+	cmake-utils_src_prepare
+}
+
+src_configure() {
+	local mycmakeargs=(
+		# as indicated in the CMakeLists.txt, the shipped documentation is generated by a custom doxygen,
+		# so let's avoid rebuilding it because it will be different
+		-DMEDFILE_BUILD_DOC=OFF
+		-DMEDFILE_BUILD_FORTRAN=$(usex fortran)
+		-DMEDFILE_BUILD_PYTHON=$(usex python)
+		-DMEDFILE_BUILD_SHARED_LIBS=ON
+		-DMEDFILE_BUILD_STATIC_LIBS=OFF
+		-DMEDFILE_BUILD_TESTS=$(usex test)
+		-DMEDFILE_DOC_DIRECTORY="${EPREFIX}"/usr/share/doc/${PF}/html   # custom var created by patches
+		-DMEDFILE_INSTALL_DOC=$(usex doc)
+		-DMEDFILE_USE_MPI=$(usex mpi)
+	)
+	cmake-utils_src_configure
+}
+
+src_install() {
+	cmake-utils_src_install
+
+	# we don't need old 2.3.6 include files
+	rm -r "${ED}"/usr/include/2.3.6 || die "failed to delete obsolete include dir"
+
+	# the optimization done in CMakeLists.txt has been disabled so
+	# we need to do it manually
+	use python && python_optimize
+
+	# Prevent test executables being installed
+	if use test; then
+		rm -r "${ED}"/usr/bin/{testc,testf,testpy} || die "failed to delete test executables"
```

Comment:
So. With `USE="-python"` this part fails, because it can't find testpy. Then with `USE="python`" the tests just fail, can you investigate, and fix/restrict them?

Reply:
Ok, I split this `rm` statement, depending on USE flags for fortran and python. In an older version, I had an `rm -f`, probably for this reason, but removed it when @a17r asked me if there's a reason for forcing this, and I don't remember. Anyway, I think the splitting is more clean.

Could you be more specific where the tests fail for `USE="python"`, I cannot reproduce this. I now tried the tests for each single use of USE flag mpi, fortran and python as well as all and none USE flags at all, using python-3.6. Does fail with a specific USE flag combination for you?

Reply:
I now checked with python-3.7 too and the tests also succeed. However, the python files install into /usr/lib64/python3.7/site-packages, like for python-3.6, while python-3.7 has moved to /usr/lib/python3.7 for multilib arches.